### PR TITLE
libvterm,neovim*: update to latest

### PR DESCRIPTION
**Pull request by casr**
#### Description

- update libvterm to 0.3.3
- update neovim to v0.9.4
- update neovim-devel to 20231106

> Note: a recent update to neovim-devel set the [minimum version of libvterm to 0.3.3](https://github.com/neovim/neovim/commit/8405649f92a8a8eb254944eca15e8b0169cbb6fb)

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.7 21G651 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/libvterm/Portfile [hunk 1]
```diff
@@ -21,11 +21,11 @@ long_description \
 
 homepage            http://www.leonerd.org.uk/code/libvterm/
 
-master_sites        ${homepage}
+master_sites                https://launchpad.net/${name}/trunk/v0.3/+download/
```
Comment (raimue) [inline devel/libvterm/Portfile:24]:
```suggestion
master_sites        https://launchpad.net/${name}/trunk/v0.3/+download/
```

File: devel/libvterm/Portfile [hunk 2]
```diff
@@ -44,4 +44,6 @@ build.env-append    CC=${configure.cc} \
 
 destroot.args       ${build.args}
 
-livecheck.regex     ${name}-(\[0-9.\]+)\\.tar
+livecheck.type              regex
+livecheck.url               http://launchpad.net/${name}/+download
+livecheck.regex             ${name}-(\[0-9.\]+)\\.tar\\.gz
```
Comment (raimue) [inline devel/libvterm/Portfile:49]:
```suggestion
livecheck.type      regex
livecheck.url       http://launchpad.net/${name}/+download
livecheck.regex     ${name}-(\[0-9.\]+)\\.tar\\.gz
```

**Comment by raimue** [comment]
Thanks for the port update! Looks good to me, I just suggested a few minor whitespace fixes. Once these are fixed, if other reviewers get to this before me, this PR can be merged.

**Comment by casr** [comment]
I've fixed those points you raised. Thanks for the review @raimue

**Review by l2dy — APPROVED** [review]

**Review by raimue — APPROVED** [review]

**Comment by raimue** [comment]
Thanks again for the port update! :+1:

**Comment by casr** [comment]
Thanks for merging it!

**Comment by vkocubinsky** [comment]
Hi, thanks for update neovim port. I recently switch to macports from brew, tried to install neovim to mac M1, and get an error

```
sudo port -t install neovim
--->  Computing dependencies for neovim
--->  Fetching archive for neovim
--->  Attempting to fetch neovim-0.9.4_0.darwin_23.arm64.tbz2 from https://fra.de.packages.macports.org/neovim
--->  Attempting to fetch neovim-0.9.4_0.darwin_23.arm64.tbz2 from https://nue.de.packages.macports.org/neovim
--->  Attempting to fetch neovim-0.9.4_0.darwin_23.arm64.tbz2 from https://cph.dk.packages.macports.org/neovim
--->  Building neovim
Error: Failed to build neovim: command execution failed
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_editors_neovim/neovim/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
Error: Processing of port neovim failed
```

Indeed there is no file neovim-0.9.4_0.darwin_23.arm64.tbz2 (but exists file with very close name neovim-0.9.4_0.darwin_22.arm64.tbz2) 


```
 curl https://fra.de.packages.macports.org/neovim/|grep arm64|grep 0.9.4
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 23401    0 23401    0     0   123k      0 --:--:-- --:--:-- --:--:--  124k
<tr><td valign="top"><img src="/icons/unknown.png" alt="[   ]"></td><td><a href="neovim-0.9.4_0.darwin_20.arm64.tbz2">neovim-0.9.4_0.darwi..&gt;</a></td><td align="right">2023-11-08 02:24  </td><td align="right">6.8M</td><td>&nbsp;</td></tr>
<tr><td valign="top"><img src="/icons/unknown.png" alt="[   ]"></td><td><a href="neovim-0.9.4_0.darwin_20.arm64.tbz2.rmd160">neovim-0.9.4_0.darwi..&gt;</a></td><td align="right">2023-11-08 02:24  </td><td align="right">512 </td><td>&nbsp;</td></tr>
<tr><td valign="top"><img src="/icons/unknown.png" alt="[   ]"></td><td><a href="neovim-0.9.4_0.darwin_21.arm64.tbz2">neovim-0.9.4_0.darwi..&gt;</a></td><td align="right">2023-11-08 00:15  </td><td align="right">6.8M</td><td>&nbsp;</td></tr>
<tr><td valign="top"><img src="/icons/unknown.png" alt="[   ]"></td><td><a href="neovim-0.9.4_0.darwin_21.arm64.tbz2.rmd160">neovim-0.9.4_0.darwi..&gt;</a></td><td align="right">2023-11-08 00:15  </td><td align="right">512 </td><td>&nbsp;</td></tr>
<tr><td valign="top"><img src="/icons/unknown.png" alt="[   ]"></td><td><a href="neovim-0.9.4_0.darwin_22.arm64.tbz2">neovim-0.9.4_0.darwi..&gt;</a></td><td align="right">2023-11-07 23:13  </td><td align="right">6.8M</td><td>&nbsp;</td></tr>
<tr><td valign="top"><img src="/icons/unknown.png" alt="[   ]"></td><td><a href="neovim-0.9.4_0.darwin_22.arm64.tbz2.rmd160">neovim-0.9.4_0.darwi..&gt;</a></td><td align="right">2023-11-07 23:13  </td><td align="right">512 </td><td>&nbsp;</td></tr>

```

**Comment by raimue** [comment]
@vkocubinsky These are attempts to download a precompiled binary archive for the port. This was not yet available from our buildbot when you tried to upgrade. This is normal behavior and not an error.

As the next step, port tried to build neovim locally. However, that build failed for some reason that would be in the mentioned main.log file.

**Comment by vkocubinsky** [comment]
thanks for clarification

**Pull request merged**